src/bindings/html/dom.js

```diff
@@ -55,11 +55,14 @@ export function translateDocument(doc, langs) {
     dispatchEvent(doc, 'DOMLocalized', langs);
   };
 
-  doc.documentElement.lang = langs[0].code;
-  doc.documentElement.dir = langs[0].dir;
-  return translateFragment.call(
-    this, doc.documentElement).then(
-      setDOMLocalized, setDOMLocalized);
+  if (langs[0].code !== doc.documentElement.getAttribute('lang')) {
+    return translateFragment.call(this, doc.documentElement).then(() => {
+        doc.documentElement.lang = langs[0].code;
+        doc.documentElement.dir = langs[0].dir;
+        setDOMLocalized();
```

Comment:
No, translateFragment error should not cause setDOMLocalized. It should never happen and not firing the event will make it easier to spot and debug